token/program2/src/processor.rs

```diff
@@ -397,14 +447,74 @@ impl Processor {
         Ok(())
     }
 
+    /// Processes a [FreezeAccount](enum.TokenInstruction.html) or a
+    /// [ThawAccount](enum.TokenInstruction.html) instruction.
+    pub fn process_toggle_freeze_account(
+        program_id: &Pubkey,
+        accounts: &[AccountInfo],
+        desired_account_state: AccountState,
```

Comment:
I think I'd prefer this to be a boolean: freeze/unfreeze, to dumb down the implementation.   `AccountState`  as an input technically allows the caller to pass in `Uninitialized` and we don't handle that (not do we need to)

Reply:
I had it as a boolean at first, but I didn't like the ambiguity of `false`.
So I replaced it with an enum (something like ToogleAction::Freeze / ToggleAction::Unfreeze), which then seemed superfluous, since the AccountState enum already essentially does this.

But, meh. Pushed the boolean.